oak_crypto/src/encryptor.rs

```diff
@@ -210,3 +219,70 @@ impl ServerEncryptor {
         Ok(response)
     }
 }
+
+/// Encryptor object for decrypting client requests that are received by the server and encrypting
+/// server responses that will be sent back to the client. Each Encryptor object corresponds to a
+/// single crypto session between the client and the server.
+/// Encryptor state is initialized after receiving an initial request message containing client's
+/// encapsulated public key.
+///
+/// Sequence numbers for requests and responses are incremented separately, meaning that there could
+/// be multiple responses per request and multiple requests per response.
+// TODO(#4311): Merge `AsyncServerEncryptor` and `ServerEncryptor` once there is `async` support in
+// the Restricted Kernel.
+pub struct AsyncServerEncryptor {
+    recipient_context_generator: Arc<dyn AsyncRecipientContextGenerator>,
+    inner: Option<ServerEncryptor>,
+}
+
+impl AsyncServerEncryptor {
+    pub fn new(recipient_context_generator: Arc<dyn AsyncRecipientContextGenerator>) -> Self {
+        Self {
+            recipient_context_generator,
+            inner: None,
+        }
+    }
+
+    /// Decrypts a [`EncryptedRequest`] proto message using AEAD.
+    /// Returns a response message plaintext and associated data.
+    /// <https://datatracker.ietf.org/doc/html/rfc5116>
+    pub async fn decrypt(
+        &mut self,
+        encrypted_request: &EncryptedRequest,
+    ) -> anyhow::Result<(Vec<u8>, Vec<u8>)> {
```

Comment:
Could you create a struct to hold the plaintext and associated data, instead of a tuple with two identical types? Probably also in the ServerEncryptor.

Reply:
Will add this in a separate PR, since it will involve changing a lot of other non-related code